combat-trainer.lic

```diff
@@ -137,7 +139,7 @@ class SetupProcess
       echo('skipping summoned weapons because no moonblade available') if $debug_mode_ct
       weapon_training = weapon_training.reject { |skill, _| game_state.summoned_info(skill) }
     end
-    new_weapon_skill = weapon_training.min_by { |skill, _| [DRSkill.getxp(skill), DRSkill.getrank(skill)] }.first
+    new_weapon_skill = weapon_training.min_by { |skill, _| [DRSkill.getxp(skill), @priority_weapons.include?(skill) ? -1 : 0, DRSkill.getrank(skill)] }.first
```

Comment:
If you do not override the `priority_weapons` setting in your character YAML, then I think `@priority_weapons` will be `undefined`. Does this line throw in that case (e.g. it tries to check `includes?` on `undefined`)?

Reply:
We use `base-empty` to prevent these kinds of problems from happening. If I'm right, you'll want to update that.

Reply:
Yeah you are definitely right. Thanks for catching that, I will update `base-empty.yaml` with an empty array. 

Out of curiosity, I did add to `base.yaml` just the tag `priority_weapons:`. I see now that `base-empty.yaml` is being used to provide an empty default value. Is the purpose of `base.yaml` to provide just a set of default keys (and sometimes sensible defaults)?

Reply:
Pretty much. `base-empty` also makes it easier for users to 'null out' a setting without having to know what the default value should be.